lhotse/cut/base.py

```diff
@@ -470,6 +472,7 @@ def trim_to_supervisions(
             if not keep_overlapping:
                 # Ensure that there is exactly one supervision per cut.
                 trimmed = trimmed.filter_supervisions(lambda s: s.id == segment.id)
+                trimmed.id = segment.id
```

Comment:
This is a bit un-intuitive that the IDs will have one format or another depending on the setting `keep_overlapping`. Maybe we could always set the cut ID to supervision ID regardless of this flag?

Reply:
Oh, good catch. There could be a single supervision even when keep_overlapping is set to True.

Reply:
I see. I didn't do overlapping ASR corpus before, and didn't want to break any pipeline.
If it's still good to set `cut_id` to `utt_id` while there is overlapping, we can revert this PR and modify it, or just open a new one.

Reply:
Feel free to open a new PR

Reply:
@wgb14 What I mean (and changed in https://github.com/lhotse-speech/lhotse/pull/855) was that even if `keep_overlapping` is set to True, it may happen that a trimmed cut has only 1 supervision, if there was no overlapping supervision originally. In such case, it makes sense to set the trimmed cut id to that of the supervision.

Now, even if the cut has multiple overlapping supervisions, one could argue that its boundaries are defined by 1 "main" supervision, and the others just happen to be included. So it would still make sense to change the cut id to that of the main supervision.

Reply:
Make sense, thanks